modguard/show.py

```diff
@@ -0,0 +1,42 @@
+import yaml
+from modguard.core.boundary import BoundaryTrie
+from typing import Any, Dict
+
+# This type hint only works on more recent versions
+# result_dict: TypeAlias = dict[str, str | bool | 'result_dict']
+
+def boundary_trie_to_dict(boundary_trie: BoundaryTrie) -> Dict[str, Any]:
```

Comment:
We may run into issues in the case where part of the file structure uses the name `is_public` or `is_boundary`, which would be weird but not that unusual.